include/simics/dmllib.h

```diff
@@ -1221,6 +1219,454 @@ _deserialize_simple_event_data(
+// follows:
+//
+//  [ s                    [i*]      [a*]          [[s[i*]]*]           ]
+//    ^ after-on-hook key  ^indices  ^arguments    ^after domains
+//                                                  (list of identities)
+//
+// The after-on-hook key is used to look up the _dml_after_on_hook_into_t
+// needed to deserialize this element.
+
+// val is a serialization of an after elem of a hook queue, minus the tag
+UNUSED static set_error_t
+_DML_deserialize_hook_queue_elem_after(ht_str_table_t *callback_ht,
+                                       ht_str_table_t *id_info_ht,
+                                       const _id_info_t *id_infos,
+                                       attr_value_t val,
+                                       _dml_hook_queue_elem_t *e) {
+    _simple_event_data_t data = { 0 };
+
+    const char *callback_key    = SIM_attr_string(SIM_attr_list_item(val, 0));
+    attr_value_t indices_attr   = SIM_attr_list_item(val, 1);
+    attr_value_t arguments_attr = SIM_attr_list_item(val, 2);
+    attr_value_t domains_attr   = SIM_attr_list_item(val, 3);
+
+    set_error_t error = Sim_Set_Ok;
+
+    _dml_after_on_hook_info_t *callback_info = ht_lookup_str(callback_ht,
+                                                             callback_key);
+    if (unlikely(!callback_info)) {
+        SIM_c_attribute_error("Failed lookup using after callback key: '%s'",
+                              callback_key);
+        error = Sim_Set_Illegal_Value;
+        goto error;
+    }
+
+    _id_info_t id_info = id_infos[callback_info->method_parent_id - 1];
+    error = _deserialize_simple_event_indices(
+        id_info.dimsizes, id_info.dimensions, &indices_attr,
+        &data.indices);
+    if (unlikely(error != Sim_Set_Ok)) {
+        goto error;
+    }
+
+    error = _deserialize_simple_event_arguments(
+        callback_info->args_size, callback_info->args_deserializer,
+        &arguments_attr, &data.args);
+    if (unlikely(error != Sim_Set_Ok)) {
+        goto error;
+    }
+
+    error = _deserialize_simple_event_domains(
+        id_info_ht, &domains_attr, &data.domains, &data.no_domains);
+    if (unlikely(error != Sim_Set_Ok)) {
+        goto error;
+    }
+
+    e->callback = callback_info->callback;
+    e->data = data;
+    e->callback_key = callback_info->callback_key;
+    return Sim_Set_Ok;
+  error:
+    _free_simple_event_data(data);
+    return error;
+}
+
+// Serializes an after elem of a hook queue, and INCLUDES the "after" tag
+UNUSED static attr_value_t
+_DML_serialize_hook_queue_elem_after(ht_str_table_t *callback_ht,
+                                     const _id_info_t *id_infos,
+                                     _dml_hook_queue_elem_t elem) {
+    _dml_after_on_hook_info_t *callback_info =
+        ht_lookup_str(callback_ht, elem.callback_key);
+    ASSERT(callback_info);
+    _id_info_t id_info = id_infos[callback_info->method_parent_id - 1];
+    attr_value_t callback_key_attr =
```

Comment:
C style says `=` after newline. (though I think it talks about binary operators, which `=` arguably is not in this case. But I think the intent of the style is clear)